modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -428,7 +431,10 @@ public class TableManager implements IgniteTablesInternal, IgniteComponent {
 
     private final CompletableFuture<Void> readyToProcessReplicaStarts = new CompletableFuture<>();
 
-    private final Map<Integer, Set<TableImpl>> tablesPerZone = new ConcurrentHashMap<>();
+    /** Mapping zone identifier to a collection of tables related to the zone. */
+    private final Map<Integer, Set<TableImpl>> tablesPerZone = new HashMap<>();
+    /** Locks to synchronize an access to the {@link #tablesPerZone}. */
+    private final Map<Integer, StampedLock> tablesPerZoneLocks = new ConcurrentHashMap<>();
```

Comment:
Where do we remove elements from the `tablesPerZoneLocks`?

Reply:
Added a comment with a link to the corresponding ticket. https://issues.apache.org/jira/browse/IGNITE-25347